helpers/tcp_client.py

```diff
@@ -0,0 +1,209 @@
+import abc
+import contextlib
+import subprocess
+from typing import Optional, Type
+
+from scapy.all import L3RawSocket, conf, sr, sr1
+from scapy.layers.inet import IP, TCP
+
+__author__ = "Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2024 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+
+class ScapyFix:
+    @classmethod
+    @abc.abstractmethod
+    def apply(cls):
+        """
+        Do some changes to fix something
+        """
+
+    @classmethod
+    @abc.abstractmethod
+    def rollback(cls):
+        """
+        Rollback made changes in apply method
+        """
+
+
+class ScapyLocalhostRequestFix(ScapyFix):
+    """
+    This fix allows to send requests to localhost using lo interface
+
+    https://scapy.readthedocs.io/en/latest/troubleshooting.html#i-can-t-ping-127-0-0-1-or-1-scapy-does-not-work-with-127-0-0-1-or-1-on-the-loopback-interface
+    """
+
+    @classmethod
+    def apply(cls) -> None:
+        conf.L3socket = L3RawSocket
+
+    @classmethod
+    def rollback(cls) -> None:
+        conf.L3socket = None
+
+
+class ScapyTCPHandshakeResetFix(ScapyFix):
+    """
+    This fix creates rules in iptables to ignore kernel RESET answer
+    because of scapy uses user space to send packages.
+
+    https://scapy.readthedocs.io/en/latest/troubleshooting.html#my-tcp-connections-are-reset-by-scapy-or-by-my-kernel
+    """
+
+    @classmethod
+    def apply(cls) -> None:
+        cmd = [
+            "iptables",
+            "-A",
+            "OUTPUT",
+            "-p",
+            "tcp",
+            "--tcp-flags",
+            "RST",
+            "RST",
+            "--sport",
+            "55555",
+            "-j",
+            "DROP",
+        ]
+        result = subprocess.run(cmd, capture_output=True, text=True)
```

Comment:
You must not use `subprocess` in our framework. We have `run_cmd` from `helpers/remote.py`